crate_universe/extension.bzl

```
@@ -290,10 +325,39 @@ _annotation = tag_class(
     ),
 )
 
+_from_specs = tag_class(
+    doc = "Generates a repo @crates from the defined `spec` tags",
+    attrs = dict(
+        name = attr.string(doc = "The name of the repo to generate", default = "crates"),
+        cargo_config = CRATES_VENDOR_ATTRS["cargo_config"],
+        generate_binaries = CRATES_VENDOR_ATTRS["generate_binaries"],
+        generate_build_scripts = CRATES_VENDOR_ATTRS["generate_build_scripts"],
+        supported_platform_triples = CRATES_VENDOR_ATTRS["supported_platform_triples"],
+    ),
+)
+
+# This should be kept in sync with crate_universe/private/crate.bzl.
+_spec = tag_class(
```

nit: can you have globals be ALL_CAPS?
```suggestion
_SPEC = tag_class(
```

---

That wouldn't match the existing `tag_class` definitions for `from_cargo` and `annotation`, and I think it would make the `module_extension` call look funky. These are basically functions and `tag_class` is sort of a function factory if you squint a little.